src/vm/customattribute.cpp

```diff
@@ -165,11 +165,11 @@ CustomAttributeManagedValues Attribute::GetManagedCaValue(CaValue* pCaVal)
         ULONG length = pCaVal->arr.length;
         BOOL bAllBlittableCa = arrayType != SERIALIZATION_TYPE_ENUM;
 
-        if (length == (ULONG)-1)
```

Comment:
As with my comments on the other function, I think it's better to keep the early out and move the NewString(...szEnumName) call above this. This requires replicating the GC_PROTECT. You don't need to replicate the "if (!bAllBlittableCa)" check as you're already checking "arrayType == SERIALIZATION_TYPE_ENUM", and that's the same check.